src/platform/plugins/shared/unused_urls_cleanup/server/plugin.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type {
+  PluginInitializerContext,
+  CoreSetup,
+  CoreStart,
+  Plugin,
+  Logger,
+} from '@kbn/core/server';
+import { TaskManagerSetupContract } from '@kbn/task-manager-plugin/server';
+import { TASK_ID, DELETE_UNUSED_URLS_TASK } from '@kbn/unused-urls-cleanup/server/constants';
+import { runDeleteUnusedUrlsTask } from '@kbn/unused-urls-cleanup/server/lib';
+import type { UnusedUrlsCleanupPluginSetup, UnusedUrlsCleanupPluginStart } from './types';
+import type { UnusedUrlsCleanupPluginConfig } from './config';
+
+export class UnusedUrlsCleanupPlugin implements Plugin {
+  private readonly logger: Logger;
+  private readonly config: UnusedUrlsCleanupPluginConfig;
+  private taskManagerSetup: TaskManagerSetupContract | undefined;
+
+  constructor(initializerContext: PluginInitializerContext) {
+    this.logger = initializerContext.logger.get();
+    this.config = initializerContext.config.get<UnusedUrlsCleanupPluginConfig>();
+  }
+
+  public setup(_core: CoreSetup, { taskManager }: UnusedUrlsCleanupPluginSetup) {
+    this.taskManagerSetup = taskManager;
+  }
+
+  public start(core: CoreStart, { taskManager }: UnusedUrlsCleanupPluginStart) {
+    const {
+      logger,
+      taskManagerSetup,
+      config: { maxAge },
+    } = this;
+
+    if (!taskManagerSetup) {
+      logger.error('taskManagerSetup is not defined');
+      return;
+    }
+
+    const savedObjectsRepository = core.savedObjects.createInternalRepository();
```

Comment:
One important thing to test is different spaces. I don't know how we store URLs from different spaces, but we need to double-check that our cleanup logic works globally across all spaces.